mmv1/products/datapipeline/DataPipeline.yaml

```
@@ -0,0 +1,400 @@
+      pipeline_name: 'my-pipeline'
+      account_id: 'my-account'
+parameters:
+  - !ruby/object:Api::Type::String
+    name: 'region'
+    description: 'A reference to the region'
+    url_param_only: true
+properties:
+  - !ruby/object:Api::Type::String
+    name: 'name'
+    description: |
+      "The pipeline name. For example': 'projects/PROJECT_ID/locations/LOCATION_ID/pipelines/PIPELINE_ID."
+      "- PROJECT_ID can contain letters ([A-Za-z]), numbers ([0-9]), hyphens (-), colons (:), and periods (.). For more information, see Identifying projects."
+      "LOCATION_ID is the canonical ID for the pipeline's location. The list of available locations can be obtained by calling google.cloud.location.Locations.ListLocations. Note that the Data Pipelines service is not available in all regions. It depends on Cloud Scheduler, an App Engine application, so it's only available in App Engine regions."
+      "PIPELINE_ID is the ID of the pipeline. Must be unique for the selected project and location."
+    required: true
+    immutable: true
+    pattern: projects/{{project}}/locations/{{region}}/functions/{{name}}
+    custom_flatten: 'templates/terraform/custom_flatten/name_from_self_link.erb'
+    custom_expand: 'templates/terraform/custom_expand/shortname_to_url.go.erb'
+  - !ruby/object:Api::Type::String
+    name: 'displayName'
+    description: |
+      The display name of the pipeline. It can contain only letters ([A-Za-z]), numbers ([0-9]), hyphens (-), and underscores (_).
+  - !ruby/object:Api::Type::Enum
+    name: 'type'
+    description: |
+      The type of the pipeline. This field affects the scheduling of the pipeline and the type of metrics to show for the pipeline.
+      https://cloud.google.com/dataflow/docs/reference/data-pipelines/rest/v1/projects.locations.pipelines#pipelinetype
+    required: true
+    values:
+      - :PIPELINE_TYPE_UNSPECIFIED
+      - :PIPELINE_TYPE_BATCH
+      - :PIPELINE_TYPE_STREAMING
+  - !ruby/object:Api::Type::Enum
+    name: 'state'
+    description: |
+      The state of the pipeline. When the pipeline is created, the state is set to 'PIPELINE_STATE_ACTIVE' by default. State changes can be requested by setting the state to stopping, paused, or resuming. State cannot be changed through pipelines.patch requests.
+      https://cloud.google.com/dataflow/docs/reference/data-pipelines/rest/v1/projects.locations.pipelines#state
+    required: true
+    immutable: true
+    values:
+      - :STATE_UNSPECIFIED
+      - :STATE_RESUMING
+      - :STATE_ACTIVE
+      - :STATE_STOPPING
+      - :STATE_ARCHIVED
+      - :STATE_PAUSED
+  - !ruby/object:Api::Type::String
+    name: 'createTime'
+    description: |
+      The timestamp when the pipeline was initially created. Set by the Data Pipelines service.
+      A timestamp in RFC3339 UTC "Zulu" format, with nanosecond resolution and up to nine fractional digits. Examples: "2014-10-02T15:01:23Z" and "2014-10-02T15:01:23.045123456Z".
+      output: true
```

```suggestion
    output: true
```

Looks like tests are failing due to this not being formatted correctly. Was being considered apart of the description.